src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.RegionBuilder.cs

```diff
@@ -313,24 +291,74 @@ public void Free()
                 {
                     for (int i = firstBlockWithoutRegion; i < region.FirstBlockOrdinal; i++)
                     {
-                        Debug.Assert(blocks[i].Region == null);
-                        blocks[i].Region = result;
+                        setRegion(blocks[i]);
                     }
 
                     firstBlockWithoutRegion = region.LastBlockOrdinal + 1;
                 }
 
                 for (int i = firstBlockWithoutRegion; i <= LastBlock.Ordinal; i++)
                 {
-                    Debug.Assert(blocks[i].Region == null);
-                    blocks[i].Region = result;
+                    setRegion(blocks[i]);
                 }
 
 #if DEBUG
                 AboutToFree();
 #endif 
                 Free();
                 return result;
+
+                void setRegion(BasicBlockBuilder block)
+                {
+                    Debug.Assert(block.Region == null);
+                    block.Region = result;
+
+                    if (anonymousFunctionsMapOpt != null)
```

Comment:
A comment on this block would help. What's the purpose of visiting statements here?